### ISSUE TITLE
Android screen reader is able to interact with content beneath `transparentModal` and `containedTransparentModal` stack

### ISSUE THREAD

**Issue by tewson** [software-mansion/react-native-screens#1204]
## Description

When a screen is stacked on top of another using the `transparentModal` or `containedTransparentModal` presentation type, Android screen reader TalkBack is still able to read and interact with the content of the screen underneath even though it's not visible.

### Screenshots

https://user-images.githubusercontent.com/456562/140495081-b8f4ad0a-e89a-4841-8617-8d3f256466de.mp4

## Steps To Reproduce

1. Install [Android Accessibility Suite](https://play.google.com/store/apps/details?id=com.google.android.marvin.talkback) to an Android device.
2. Run this project's [example app](https://github.com/software-mansion/react-native-screens/tree/master/Example) on the device.
3. [Turn on TalkBack](https://support.google.com/accessibility/android/answer/6007100).
4. In the example app, open the `containedTransparentModal` example modal.
5. Tap within the modal area.

### Expected behavior

Only the modal content is read by TalkBack.

### Actual behavior

TalkBack can read and interact with other example buttons underneath the modal.

## Reproduction

The issue is reproducible from the example app from this repository: https://github.com/software-mansion/react-native-screens/tree/master/Example.

## Platform

- [ ] iOS
- [x] Android
- [ ] Web
- [ ] Windows
- [ ] tvOS

## Workflow

- [ ] Managed workflow
- [x] Bare workflow

## Package versions

| package                                | version |
| -------------------------------------- | ------- |
| react-native                           |   0.66.0     |
| @react-navigation/native               |     5.9.2    |
| react-native-screens                   |     3.9.0    |
